tasks/eth/mode-001-MCP-L1/SignFromJson.s.sol

```diff
@@ -0,0 +1,416 @@
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {AddressManager} from "@eth-optimism-bedrock/src/legacy/AddressManager.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {ISemver} from "@eth-optimism-bedrock/src/universal/ISemver.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {EIP1967Helper} from "@eth-optimism-bedrock/test/mocks/EIP1967Helper.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "solady/utils/LibString.sol";
+
+// Interface used to read various data from contracts. This is an aggregation of methods from
+// various protocol contracts for simplicity, and does not map to the full ABI of any single contract.
+interface IFetcher {
+    function overhead() external returns (uint256); // SystemConfig
+    function scalar() external returns (uint256); // SystemConfig
+    function guardian() external returns (address); // SuperchainConfig
+    function L2_BLOCK_TIME() external returns (uint256); // L2OutputOracle
+    function SUBMISSION_INTERVAL() external returns (uint256); // L2OutputOracle
+    function FINALIZATION_PERIOD_SECONDS() external returns (uint256); // L2OutputOracle
+    function startingTimestamp() external returns (uint256); // L2OutputOracle
+    function startingBlockNumber() external returns (uint256); // L2OutputOracle
+    function owner() external returns (address); // ProtocolVersions
+    function required() external returns (uint256); // ProtocolVersions
+    function recommended() external returns (uint256); // ProtocolVersions
+}
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    // Chains for this task.
+    string constant l1ChainName = "mainnet";
+    string constant l2ChainName = "mode";
+
+    // Known EOAs to exclude from safety checks.
+    address constant l2OutputOracleProposer = 0x674F64D64Ddc198db83cd9047dF54BF89cCD0ddB; // cast call $L2_OUTPUT_ORACLE_PROXY "PROPOSER()(address)" # https://github.com/ethereum-optimism/superchain-registry/blob/4c005f16ee1b100afc08a35a2e418d849bea044a/superchain/extra/addresses/mainnet/mode.json#L7
+    address constant l2OutputOracleChallenger = 0x309Fe2536d01867018D120b40e4676723C53A14C; // https://github.com/ethereum-optimism/superchain-registry/blob/4c005f16ee1b100afc08a35a2e418d849bea044a/superchain/extra/addresses/mainnet/mode.json#L14
+    address constant systemConfigOwner = 0x4a4962275DF8C60a80d3a25faEc5AA7De116A746; // https://github.com/ethereum-optimism/superchain-registry/blob/4c005f16ee1b100afc08a35a2e418d849bea044a/superchain/extra/addresses/mainnet/mode.json#L12 
```

Comment:
l2OutputOracleChallenger and systemConfigOwner aren't EOA's so we should move their variable declaration to a different comment header